### ISSUE TITLE
Sign Official Godot Releases

### ISSUE THREAD

**Issue by hendursaga** [godotengine/godot-proposals#6042]
### Describe the project you are working on

N/A

### Describe the problem or limitation you are having in your project

I would like to have extra confidence that the Godot infrastructure has not been compromised.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Some form of signing for the SHA512 checksums, whether that would be PGP, signify, SSH, or other. 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

This was [already proposed some years back](https://github.com/godotengine/godot/issues/3989), but it looks like nothing came of it. In particular, [this comment](https://github.com/godotengine/godot/issues/3989#issuecomment-193946793) has some good suggestions.

Personally, for a project like this, I'd be more in favor of using signify or SSH keys, as they're a lot simpler than PGP, and I don't think many if anyone would be taking advantage of the Web of Trust anyways.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

N/A

### Is there a reason why this should be core and not an add-on in the asset library?

It concerns the security of the Godot project.

**Comment by YuriSizov** [comment]
This is for Linux specifically, right? Godot's executable for macOS and Windows are already signed, because that's practically a requirement for those platforms.

**Comment by hendursaga** [comment]
> This is for Linux specifically, right?

No, I wouldn't say so, considering you already publish the checksums for everything already..? All you would need to do is sign the checksums file and it covers everything.

**Comment by Calinou** [comment]
One issue with signing is that it tends to make the release process more cumbersome, especially as the key needs to be guarded securely (which may involve using hardware security tokens and the like). Considering the amount of manual effort required to publish even a small patch release, it may not be worth the trouble just yet.

**Comment by hendursaga** [comment]
> it may not be worth the trouble just yet.

Oh, I was thinking of introducing it for when the big 4.0 release happens, not necessarily retroactively..
